File: src/main/java/com/somefrills/events/PlaceBlockEvent.java
package com.somefrills.events;

import net.minecraft.block.Block;
import net.minecraft.item.ItemPlacementContext;

public class PlaceBlockEvent extends Cancellable {
    public final ItemPlacementContext context;
    public final Block block;

    public PlaceBlockEvent(ItemPlacementContext context, Block block) {
        this.context = context;
        this.block = block;
    }
}

File: src/main/kotlin/com/somefrills/events/PlaceBlockEvent.kt
package com.somefrills.events

import net.minecraft.world.item.context.BlockPlaceContext
import net.minecraft.world.level.block.Block
import com.somefrills.events.FrillsEvent.Cancellable

class PlaceBlockEvent(@JvmField val context: BlockPlaceContext, val block: Block) : FrillsEvent(), Cancellable
